FreeRTOS/FreeRTOS
# FreeRTOS+TCP : renewing DHCP lease while network is down

Description
-----------
This is a response to [issue 1874](https://github.com/aws/amazon-freertos/issues/1874).
Once the DHCP lease time expires, a new DHCP session will be started. It doesn' t make sense to start a new discovery as long as the network is down.

This patch will test if the network is up before starting a new DHCP request.

Symbolically:
~~~c
   if( FreeRTOS_IsNetworkUp() )
   {
        /* Resend the request at the appropriate time to renew the lease. */
    }
    else
    {
        /* Wait 5 more seconds and check again. */
    }
~~~

Test Steps
-----------
Please apply the patch presented in [issue 1874](https://github.com/aws/amazon-freertos/issues/1874):
~~~c
    WIFI_Disconnect();
    vTaskDelay( portMAX_DELAY );
~~~
Related Issue
-----------
[issue 1874](https://github.com/aws/amazon-freertos/issues/1874).


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
